src/action.js

```diff
@@ -290,6 +290,16 @@ class Action extends Emitter {
       }
     }
   }
+
+  toJSON() {
+    return {
+      id: this.id,
+      status: this.status,
+      type: this.type,
+      payload: this.payload,
+      children: this.children
```

Comment:
Children is. JSON.stringify recursively calls `toJSON` on properties (and so on). Payload actually is an interesting one. It might not be serializable.